Add `pg_pin`s to Liberty file output

infinitymdm opened the pull request:
Fixes #72. Adds pg_pin write to cell outputs. Also adds a characterization run (just a GF180 inverter) to the test workflow.

Long term, we probably need to refactor the whole `named_nodes` construct. Currently `pg_pin`s are really awkward to add to cells because they're stored in CharacterizationSettings. (There's actually quite a bit I'd like to rearchitect here, but that's a longer term task)

lrburle commented on charlib/liberty/cell.py:
I really like your use of the `__str__` dunder method for the `PgPin` class here. Makes it really slick and easy to return pin information for unique instances.

lrburle reviewed (approved):
Nothing here seems out of place. I would only advise to add variable type hinting to all of your method definitions. You have done it to a good portion here but adding them to the rest of the methods make it easier to read and follow. Other than that slight nit pick, I think everything looks good. Good work!

infinitymdm commented:
@lrburle Thanks for the review! Fixed the missing type hinting and changed the `valid_*` methods to classmethods, so they're accessible without an instance. Looks like tests are passing, including characterization runs, so merging this immediately.
